### DISCUSSION TITLE
Neogit mappings

### DISCUSSION THREAD

**Discussion by atimofeev** [nix-community/nixvim#1172] [Q&A] [answered]
I've tried many different ways, but I can't get custom mappings working for Neogit setup. 
Unfortunately, documentation has no examples and reading through code gives me no hints on how to configure it.

This is my neovim config for neogit mappings:
```lua
mappings = {
  commit_editor = {
    ["q"] = "Close",
    ["<M-c>"] = "Submit",
    ["<M-k>"] = "Abort",
  },
  popup = {
    ["p"] = "PushPopup",
    ["P"] = "PullPopup",
  },
},
```
How to do this correctly?
Any help is appreciated.

**Comment by GaetanLepage** [comment, marked as answer]
Hey !
I just merged a rewrite of the `plugins.neogit` module.
This should be possible.
You can find an example in the tests: https://github.com/nix-community/nixvim/blob/8c0bdb17e980b2ca64860e309ce601dd92ce0c51/tests/test-sources/plugins/git/neogit.nix#L139-L215

**Reply by atimofeev** [reply]
Works perfectly now!
```nix
    neogit = {
      enable = true;
      settings = {
        auto_refresh = true;
        disable_hint = true;
        disable_signs = true;
        mappings = {
          commit_editor = {
            q = "Close";
            "<M-c>" = "Submit";
            "<M-k>" = "Abort";
          };
          popup = {
            p = "PushPopup";
            P = "PullPopup";
          };
        };
      };
    };
```

Thank you for such a swift answer and implementation!

**Reply by GaetanLepage** [reply]
Cool ! You're welcome :)
